json_test.go

```diff
@@ -17,644 +19,800 @@ var _ = Describe("JSON Commands", Label("json"), func() {
 	ctx := context.TODO()
 	var client *redis.Client
 
-	BeforeEach(func() {
-		client = redis.NewClient(&redis.Options{Addr: ":6379"})
-		Expect(client.FlushAll(ctx).Err()).NotTo(HaveOccurred())
-	})
-
-	AfterEach(func() {
-		Expect(client.Close()).NotTo(HaveOccurred())
-	})
-
-	Describe("arrays", Label("arrays"), func() {
-		It("should JSONArrAppend", Label("json.arrappend", "json"), func() {
-			cmd1 := client.JSONSet(ctx, "append2", "$", `{"a": [10], "b": {"a": [12, 13]}}`)
-			Expect(cmd1.Err()).NotTo(HaveOccurred())
-			Expect(cmd1.Val()).To(Equal("OK"))
-
-			cmd2 := client.JSONArrAppend(ctx, "append2", "$..a", 10)
-			Expect(cmd2.Err()).NotTo(HaveOccurred())
-			Expect(cmd2.Val()).To(Equal([]int64{2, 3}))
+	setupRedisClient := func(protocolVersion int) *redis.Client {
+		return redis.NewClient(&redis.Options{
+			Addr:          "localhost:6379",
+			DB:            0,
+			Protocol:      protocolVersion,
+			UnstableResp3: true,
```

Comment:
Are we using unstable flag because of test cases with JSON + Search or it's needed for JSON standalone?

Reply:
Yes, only for JSON + SEARCH test